src/backend/marsha/core/tests/test_models_video.py

```diff
@@ -333,3 +345,467 @@ def test_model_video_set_recording_slice_manifest_key(self):
+        self,
+    ):
+        """
+        Check get_list_timestamps_attendences builds an array of timestamp
+        depending of the duration of the video and if live is still running
+        that the timestamp list is cached. To simulate that the cache is over,
+        the key is regenerated, we control when it is the case, the list is
+        regenerated.
+        """
+        # video ends one hour later
+        started = 1600000000
+        one_hour = 3600
+
+        video = VideoFactory(
+            id="a1a21411-bf2f-4926-b97f-3c48a124d528",
+            live_state=RUNNING,
+            live_info={
+                "medialive": {
+                    "input": {
+                        "id": "medialive_input_1",
+                        "endpoints": [
+                            "https://live_endpoint1",
+                            "https://live_endpoint2",
+                        ],
+                    },
+                    "channel": {"id": "medialive_channel_1"},
+                    "request_ids": ["7954d4d1-9dd3-47f4-9542-e7fd5f937fe6"],
+                },
+                "mediapackage": {
+                    "id": "mediapackage_channel_1",
+                    "endpoints": {
+                        "hls": {
+                            "id": "endpoint1",
+                            "url": "https://channel_endpoint1/live.m3u8",
+                        },
+                    },
+                },
+                "cloudwatch": {"logGroupName": "/aws/lambda/dev-test-marsha-medialive"},
+                "started_at": started,
+            },
+            live_type=RAW,
+        )
+        cache_key = f"video:{video.id}:get_list_timestamps_attendences"
+        date_now = datetime.fromtimestamp(started + one_hour)
+        with mock.patch.object(timezone, "now", return_value=date_now):
+            list_timestamp_ori = {
+                "1600000000": {},
+                "1600001200": {},
+                "1600002400": {},
+                "1600003600": {},
+            }
+            self.assertEqual(video.live_duration, one_hour)
+            self.assertEqual(video.live_ended_at, started + one_hour)
+            self.assertEqual(
+                video.get_list_timestamps_attendences(), list_timestamp_ori
+            )
+
+        # duration and live_ended_at property change as live is still running,
+        # get_list_timestamps_attendences don't as the list is cached
+        date_now = datetime.fromtimestamp(started + one_hour + 3)
+        with mock.patch.object(timezone, "now", return_value=date_now):
+            self.assertEqual(video.live_duration, one_hour + 3)
+            self.assertEqual(video.live_ended_at, started + one_hour + 3)
+            self.assertEqual(
+                video.get_list_timestamps_attendences(), list_timestamp_ori
+            )
+            self.assertEqual(cache.get(cache_key), list_timestamp_ori)
+
+        # reinit key of cache
+        cache.touch(cache_key, timeout=0)
```

Comment:
Instead of computing a cache key that is subject to change in the time, you can reset the cache `cache.clear()`